build.sbt

```diff
@@ -181,11 +181,71 @@ addCommandAlias(
   "+publishLocal; metals/runMain scala.meta.metals.DownloadDependencies "
 )
 
-commands += Command.command("save-expect") { s =>
-  "unit/test:runMain tests.SaveExpect" ::
-    s
+def configureMtagsScalaVersionDynamicaly(
+    state: State,
+    scalaV: String,
+    projectV: Option[String] = None
+): State = {
+  val scalaVersionSettings =
+    List(
+      mtest / scalaVersion := scalaV,
+      mtags / scalaVersion := scalaV,
+      cross / scalaVersion := scalaV
+    )
+  val versionSettings =
+    projectV.map(v => mtags / version := v)
+
+  val extracted = Project.extract(state)
+  extracted
+    .appendWithSession(
+      scalaVersionSettings ++ versionSettings,
+      state
+    )
 }
 
+commands ++= Seq(
+  Command.command("save-expect") { s =>
+    "unit/test:runMain tests.SaveExpect" ::
+      s
+  },
+  Command.single("test-mtags-dyn") { (state, scalaV) =>
+    val configured = configureMtagsScalaVersionDynamicaly(state, scalaV)
+    val (out, _) =
+      Project
+        .extract(configured)
+        .runTask(cross / Test / test, configured)
+    out
+  },
+  Command.args("publish-mtags-dyn", "<scala-version> <version>") {
+    (state, args) =>
+      val configured =
+        configureMtagsScalaVersionDynamicaly(state, args(0), Some(args(1)))
+      val (out, _) =
+        Project
+          .extract(configured)
+          .runTask(mtags / PgpKeys.publishSigned, configured)
+      out
+  },
+  /**
+   * In case if this command was called from tag that matches
+   * "mtags_v$metalsVersion_$scalaVersion" then publish only mtags with these parameters
+   * Otherwise, call std `ci-release` and publish docs
+   */
+  Command.command("gh-actions-release") { s =>
+    val isTag = Option(System.getenv("GITHUB_REF"))
+      .exists(_.startsWith("refs/tags"))
+
+    val mtagsTagRegex = "^mtags_v(\\d+\\.\\d+\\.\\d+)_(.+)".r
+
+    Option(System.getenv("GITHUB_REF_NAME")) match {
+      case Some(mtagsTagRegex(metalsVersion, scalaV)) if isTag =>
+        s"publish-mtags-dyn $scalaV $metalsVersion" :: s
```

Comment:
Could we add some validation on the other clause too? So that we don't end up with broken release in case of a mistake?

Reply:
Do you mean to add validation for `"ci-release" :: "docs/docusaurusPublishGhpages" :: s` ?
I've checked CIRelease plugin - there is no requirements. It checks internally if it's a tag or snapshot.

I see that the worst thing that might happen is that if there will be a typo in tag name for mtags release it will trigger usual release with a creepy version name.

Reply:
added!